tinygrad/codegen/uopgraph.py

```diff
@@ -477,9 +481,28 @@ def delete_redundant_gates(root:UOp) -> Optional[UOp]:
   def find_gate(x:UOp) -> Optional[UOp]:
     if x.op is UOps.IF: return x
     return next((ret for s in x.src if (ret:=find_gate(s)) is not None), None)
-  if len(root.src) == 3 or (gate:=find_gate(root)) is None or gate.src[0] is not root.src[3]: return None
+  if len(root.src) == 3 or (gate:=find_gate(root)) is None or gate.src[0] is not root.src[3] \
+    and not ((sub_gate := find_gate(root.src[2])) is not None and sub_gate.src[0] is root.src[-1].src[0]):
```

Comment:
Unsure about the best way to do this. I had it as a nested if statement before, but then `return UOp(UOps.STORE, root.dtype, root.src[:3], root.arg)` was getting repeated again 🤷 This way only adds 2 lines (could be only 1 if we want to bump the `return None` to this line) but is maybe a bit harder to read.

Reply:
what is the most readable and simplest you can make this?
I can't read this tbh.